ballerina-shell/README.md

```diff
@@ -0,0 +1,130 @@
+  in closures will not be reflected after the execution. The changes will be visible only for the scope belonging to the
+  snippet where the closure was defined.
+
+  ```ballerina
+  int x = 10
+  var f = function () { x = 12; }
+  f()
+  x   // <- this should output 12 but will output 10 instead
+  ```
+
+- **Enum definition with explicit expression is not fully supported** - Enum declarations with explicit expressions are
+  not supported fully. Some functionalities may not work properly. Following is an example.
+
+  ```ballerina
+  // This will not work due to type EN converting to english
+  enum Language {EN="english", TA="tamil", SI="sinhala"}
+  EN variable = "english"  // <- Error
+  ```
+
+- **Support with external libraries not tested completely** - Even though it is possible to run this against external libraries, this is not fully tested. Some issues, especially relating to invincible types, may occur. For an example following is a possible problematic scenario.
+
+  ```ballerina
+  import abc/pqr
+  // If pqr:call() returns a value of which the type is private, following will fail.
+  var r = pqr:call()
+  ```
+
+### Related Issues
+
+- **Enum type signature should be the type symbol name** - If an enum is defined with expression values, enum elements'
+  type signature would be the expression. However, this should instead be the original enum name.
+
+  ```ballerina
+  enum Language {EN="english", TA="tamil", SI="sinhala"}
+  // Type signature of variable would be `english` but should instead be `EN`.
+  EN variable = "english" 
+  ```
+
+- **Type signature of arrays with a fixed length should reflect the length** - Any array definition with a fixed length
+  should have the type signature that reflects the length. But this is not the case, as a result, array bindings are not
+  possible.
+
+  ```ballerina
+  int[3] variable = [1, 2, 3]
+  // Type signature of variable would be `int[]` but should instead be `int[3]`.
+  // As a result following is not possible.
+  [int, int, int] [a, b, c] = variable
+  ```
+
+### Notes
+
+- **The parser is imperfect** - Current parser is imperfect and is sometimes unable to detect the type of statement.
+  Please file an issue if you come across any wrong categorization of a snippet. The parser is also relatively slow
+  compared to the compilation phase, acting as a bottle-neck. So a timeout is employed to stop invalid statement parsing
+  from taking too much time. However, this might cause issues in some old hardware where the execution might be slower
+  than expected (where even valid executions might exceed the timeout).
+
+- **Type guards for global variables do not work** - Since all the variables defined in REPL top level act as global
+  variables, type guards won't work. To remedy this you can explicitly cast the variable to the required type. However,
+  note that this is the expected behavior and not a bug. This will be fixed once the language supports global type
+  guards.
```

Comment:
From the language POV, AFAIK this will not change. Since global variables can be updated in different functions, we cannot guarantee that a type guard will hold.

I think we should move this to "Known Issues" and just mention that type guards won't work because everything is defined as a global variable.

Reply:
OK